docs/Windows.md

```diff
@@ -61,7 +61,19 @@ You will need the ICU libraries from [ICU - International Components for Unicode
 
 #### Building Swift code
 
-You should use a CMake project to build a Swift program. An example CMake project with support for mixed-language support is available at [https://github.com/compnerd/swift-cmake-demo](https://github.com/compnerd/swift-cmake-demo), sample commands that build the project are documented there. The make commands are to be executed from the Native Tools Command Prompt. Note that the cmake executable to be used is the one mentioned above. (Try `cmake -version` to see which CMake version is actually being called.)
+You should use a CMake project to build a Swift program. As an example CMake project use the "HelloWorld-CMake" example from [https://github.com/compnerd/swift-cmake-demo](https://github.com/compnerd/swift-cmake-demo) and use the following commands from within the project directory to build the project for Windows:
+
+```cmd
+SET INSTALLATION_DIR=C:\Library
+SET SDK=%INSTALLATION_DIR%\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk
+SET OS=windows
+cmake -G Ninja -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=YES -D CMAKE_Swift_FLAGS="-sdk %SDK% -I %SDK%/usr/lib/swift -L %SDK%/usr/lib/swift/%OS%"
+cd build
+ninja
+ninja test
+```
+
+Here, the variables INSTALLATION_DIR, SDK, and OS are just added for clarity of the subsequent command (as used here, they they should not contain spaces). Execute these commands from the Native Tools Command Prompt (or set needed paths before executing). Note that the cmake executable to be used is the one mentioned above. (Try `cmake -version` to see which CMake version is actually being called.)
```

Comment:
I think that we should be resilient to spaces in the path.  If that causes problems we should address them.

Reply:
Using the Windows command prompt here and using CMAKE_Swift_FLAGS with some parameters for the Swift compiler poses a problem when using spaces, and I have not been able to resolve that. The problem does not exist when you use a tool (editor, IDE) where you specify each argument separately:

~~~
"cmake",
...
"-D",
"CMAKE_Swift_FLAGS=-sdk \"" + SDK + "\" -I \"" + SDK + "/usr/lib/swift\" -L \"" + SDK + "/usr/lib/swift/" + OS + "\""
~~~

That works. So maybe we should just leave the command line as it is, I added "...better use a tool where you can define each argument to cmake separately" in the description. (We could make a little program that does that, but maybe this goes too far.)

Reply:
New text changed to "...better use a tool where you can define each argument to cmake separately and then put the paths inside the CMAKE_Swift_FLAGS argument in quotes".

Reply:
...and added "...instead of the outer quotes".

Reply:
... "...each path...", not "...the paths...".

Reply:
Oh oh... Incomplete testing in my side. "Should" have worked, but:

~~~
Parameters:
cmake
C:\WITH SPACES IT IS\Library\HelloWorld-CMake
-G
Ninja
-B
C:\WITH SPACES IT IS\Library\HelloWorld-CMake/build
-DCMAKE_BUILD_TYPE=RelWithDebInfo
-DBUILD_TESTING=YES
-D
CMAKE_Swift_FLAGS=-sdk "C:\WITH SPACES IT IS\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk" -I "C:\WITH SPACES IT IS\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk/usr/lib/swift" -L "C:\WITH SPACES IT IS\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk/usr/lib/swift/windows"
~~

Error:

~~~
CMake Error: The source directory "C:/WITH SPACES IT IS/Library/IS/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk/usr/lib/swift/windows" does not exist.
~~~

Hmm. This path is complete nonsense.

Reply:
...Using single quotes instead helps a little (CMAKE_Swift_FLAGS=-sdk 'C:\...), but then the compiler test fails:

~~~
 The Swift compiler

    "C:/WITH SPACES IT IS/Library/Developer/Toolchains/unknown-Asserts-development.xctoolchain/usr/bin/swiftc.exe"

  is not able to compile a simple test program.

...

<unknown>:0: error: no such file or directory: 'SPACES'
    <unknown>:0: error: no such file or directory: 'IT'
    <unknown>:0: error: no such file or directory: 'IS\Library\Developer\Platforms\Windows.platform\Developer\SDKs\Windows.sdk''
    <unknown>:0: error: no such file or directory: 'SPACES'
    <unknown>:0: error: no such file or directory: 'IT'
~~~

Reply:
...So changed text again to "...they should not contain spaces". I think we should publish it as it is, the space thing is a separate issue – it should be resolved, but the current changes to the Windows.md should not be hold back.

Reply:
I think that you need to use the correct escape character: ^ not \ like Unix.